### PR TITLE
VSP-1146 [iOS] Add button in Archive Settings that takes you to the Archive Legacy Planning

### PR DESCRIPTION
**Pull request by luciancerbu-vsp**
As a user I should be able to setup my archive legacy planning by going into the left menu, expanding the Archive Settings and Tapping on the Configure Legacy Steward.

### CODE REVIEW

File: Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift [hunk 1]
```diff
@@ -16,11 +16,18 @@ class LegacyPlanningLoadingViewController: BaseViewController<LegacyPlanningView
         
         navigationItem.title = "Legacy Planning".localized()
         
-        if viewModel?.getLegacyPlanningAccount() == true {
-            showStatus()
-        } else {
-            showIntro()
-        }
+        viewModel?.getLegacyPlanningAccount(completion: { [weak self] result in
```
Comment (flaviuvsp) [inline Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift:19]: Should we have a loading spinner here until it gets the steward?

File: Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift [hunk 2]
```diff
@@ -16,11 +16,18 @@ class LegacyPlanningLoadingViewController: BaseViewController<LegacyPlanningView
...
+            switch result {
+            case .success(let haveLegacySteward):
+                if let haveLegacySteward = haveLegacySteward, haveLegacySteward {
```
Comment (flaviuvsp) [inline Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift:22]: if haveLegacySteward != nil would be easier as you don't use the haveLegacySteward. Also, is this a bool or an object? If it's an object you should name it something else like: steward

File: Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningStewardViewController.swift [hunk 3]
```diff
@@ -86,7 +86,9 @@ class LegacyPlanningStewardViewController: BaseViewController<LegacyPlanningView
         overlayView.alpha = 0
         
         titleLabelSetup()
-        backButtonSetup()
+        if let viewControllersCount = navigationController?.viewControllers.count, viewControllersCount > 1 {
```
Comment (flaviuvsp) [inline Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningStewardViewController.swift:89]: wouldn't it work like
if navigationController?.viewControllers.count > 1 {
backButtonSetup()
}

File: Permanent/ViewControllers/SideMenuViewController.swift [hunk 4]
```diff
@@ -82,14 +88,23 @@ class SideMenuViewController: BaseViewController<AuthViewModel> {
     }
     
     func updateLeftSideMenu() {
+        var menuIndexPaths: [IndexPath] = []
+        var menuTitles: [DrawerOption] = []
+        if let selectedArchivePermission = AuthenticationManager.shared.session?.selectedArchive?.permissions(), selectedArchivePermission.contains(.legacyPlanning) {
```
Comment (flaviuvsp) [inline Permanent/ViewControllers/SideMenuViewController.swift:93]: This logic should happen in viewModel. The viewController should just call a method like viewModel.hasLegacyPermissions()

File: Permanent/ViewControllers/SideMenuViewController.swift [hunk 5]
```diff
@@ -82,14 +88,23 @@ class SideMenuViewController: BaseViewController<AuthViewModel> {
...
+            menuIndexPaths = [IndexPath(item: 1, section: 2), IndexPath(item: 2, section: 2), IndexPath(item: 3, section: 2)]
+            menuTitles = [DrawerOption.manageTags, DrawerOption.manageMembers, DrawerOption.legacyPlanning]
```
Comment (flaviuvsp) [inline Permanent/ViewControllers/SideMenuViewController.swift:95]: Looks like duplicated code. You could have: 
var menuIndexPaths: [IndexPath] = [IndexPath(item: 1, section: 2), IndexPath(item: 2, section: 2)]
var menuTitles: [DrawerOption] = [DrawerOption.manageTags, DrawerOption.manageMembers]
if viewModel.hasLegacyPermissions() {
menuIndexPaths.append(IndexPath(item: 3, section: 2))
menuTitles.append(DrawerOption.legacyPlanning)
}

File: Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift (hunk 1 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/LegacyPlanning/ViewController/LegacyPlanningLoadingViewController.swift:19]: In the design of this screen, there is no loading spinner.

File: Permanent/ViewControllers/SideMenuViewController.swift (hunk 5 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/SideMenuViewController.swift:95]: I will make this change.

File: Permanent/ViewControllers/SideMenuViewController.swift (hunk 4 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/SideMenuViewController.swift:93]: I will add this to viewModel.

**Review by flaviuvsp — APPROVED** [review]

**Pull request merged**